cirq/experiments/qubit_characterizations.py

```diff
@@ -0,0 +1,544 @@
+        angle and the second item being the corresponding excited state
+        probability.
+        """
+        return [(angle, prob) for angle, prob in zip(self._rabi_angles,
+                                                     self._excited_state_probs)]
+
+    def plot(self) -> None:
+        """Plots excited state probability vs the Rabi angle (angle of rotation
+        around the x-axis).
+        """
+        fig = pyplot.figure()
+        pyplot.plot(self._rabi_angles, self._excited_state_probs, 'ro-',
+                    figure=fig)
+        pyplot.xlabel(r"Rabi Angle ($\pi$)")
+        pyplot.ylabel('Excited State Probability')
+
+
+class RBResult(object):
+    """Class for storing and plotting data from a randomized benchmarking
+    experiment.
+    """
+
+    def __init__(self, num_cfds_seq: Sequence[int],
+                 gnd_state_probs: Sequence[float]):
+        """
+        Args:
+            num_cfds_seq: The different numbers of Cliffords in the RB study.
+            gnd_state_probs: The corresponding average ground state
+                probabilities.
+        """
+        self._num_cfds_seq = num_cfds_seq
+        self._gnd_state_probs = gnd_state_probs
+
+    @property
+    def data(self) -> Sequence[Tuple[int, float]]:
+        """Returns a sequence of tuple pairs with the first item being a
+        number of Cliffords and the second item being the corresponding average
+        ground state probability.
+        """
+        return [(num, prob) for num, prob in zip(self._num_cfds_seq,
+                                                 self._gnd_state_probs)]
+
+    def plot(self) -> None:
+        """Plots the average ground state probability vs the number of
+        Cliffords in the RB study.
+        """
+        fig = pyplot.figure()
+        pyplot.plot(self._num_cfds_seq, self._gnd_state_probs, 'ro-',
+                    figure=fig)
+        pyplot.xlabel(r"Number of Cliffords")
+        pyplot.ylabel('Ground State Probability')
+
+
+class TomographyResult(object):
+    """Class for storing and plotting a density matrix obtained from a state
+    tomography experiment."""
+
+    def __init__(self, density_matrix: numpy.ndarray):
+        """
+        Args:
+            density_matrix: The density matrix obtained from tomography.
+        """
+        self._density_matrix = density_matrix
+
+    @property
+    def data(self) -> numpy.ndarray:
+        """Returns an n^2 by n^2 complex matrix representing the density
+        matrix of the n-qubit system.
+        """
+        return self._density_matrix
+
+    def plot(self) -> None:
+        """Plots the real and imaginary parts of the density matrix as two
+        3D bar plots.
+        """
+        _plot_density_matrix(self._density_matrix)
+
+
+def rabi_oscillations(sampler: sim.SimulatesSamples, qubit: devices.GridQubit,
+                      final_angle: float, num_shots: int,
+                      num_points: int) -> RabiResult:
+    """
```

Comment:
Set summary line of docstring (the one inline with the """) to `Runs a Rabi oscillation experiment.`. Empty line after that.

Reply:
Done